README.rst

```diff
@@ -180,18 +170,12 @@ Organizations using PUDL
 
 .. _pudl_orgs_start:
 
-This is a partial list of organizations that have used PUDL in their work.
```

Comment:
Even though it is duplicative, I think we want to leave this here - most people won't follow the link, and it's good to highlight our partners in multiple places.